### ISSUE TITLE
Enable use of tf.metrics functions in model.fit

### ISSUE THREAD

**Issue by tafsiri** [tensorflow/tfjs#1171]
#### TensorFlow.js version
Any

#### Browser version
Any

#### Describe the problem or feature request

Summarizing a discussion with @caisq.

We want the following function to work.

```js
model.compile({
  optimizer: tf.train.sgd(LEARNING_RATE),
  loss: tf.losses.meanSquaredError,
  metrics: [tf.metrics.meanSquaredError],
});
```

It does currently work however the logs object ends up with a key that has the entire serialized function as the key (like in the scrreenshot below)

<img width="1293" alt="screen shot 2019-01-29 at 1 54 46 pm" src="https://user-images.githubusercontent.com/26408/51932618-7f266000-23cd-11e9-94ab-c359903e1a92.png">

Ideally we would want the logs object to use the name of the function (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name), in this case 'meanSquaredError'. However We need to take care of function name changes during minification so we should look into explicitly attaching a `name` property to the function as they are being defined (or a `__name` if we find those get mangled).

#### Alternatives

The currently alternative is to use shortcut strings like `mse` as a metric. However these are not all documented and it would be hard to keep the documentation of them in sync with the code if we just added a table of these shortcut strings in model.fit documentation. Also users may still wonder about what they can do with the `tf.losses.*` namespace and `tf.metrics.*` namespace if we never use them in our code samples. The approach above just lets us link to those sections of the docs when saying what the allowed losses and metrics are. 

We would also be able to document the shorthand string alongside the function entry in the docs.

**Comment by syt123450** [comment]
Hi @tafsiri @caisq , I create a PR [tensorflow/tfjs-layers#551](https://github.com/tensorflow/tfjs-layers/pull/551) which makes the function to work.
```js
model.compile({
  optimizer: tf.train.sgd(LEARNING_RATE),
  loss: tf.losses.meanSquaredError,
  metrics: [tf.metrics.meanSquaredError],
});
```
<img width="610" alt="Screen Shot 2019-05-31 at 3 09 30 AM" src="https://user-images.githubusercontent.com/7977100/58701795-cb23ea00-8358-11e9-84e2-4cf748dc525a.png">

Could you please take a look at it? Thanks.

**Comment by tafsiri** [comment]
Closed in https://github.com/tensorflow/tfjs-layers/pull/551
